src/cache/asset.rb

```diff
@@ -25,8 +25,16 @@ def asset_for_node(node)
         data[:node][node.name.to_sym]
       end
 
-      def unassign_asset(asset_name)
-        data.delete_if { |_key, value| value == asset_name }
+      def unassign_asset(asset_name, *node_name)
```

Comment:
What is the purpose of accepting an array of `node_names`? Firstly, as it is an array, it should be plural. Secondly, the condition below only every checks the first element of the array.

AFAICS, only the first node will be unassigned and the rest will be ignored. Either accept and array and unassign all the nodes in it OR remove the array input altogether.

If all you want is an optional input, you can use; `node_name = nil`.

Reply:
Yep only wanted an optional input